pkg/config/config.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	envPrefix = "CHATTERINO_API"
+	appName   = "chatterino-api"
+)
+
+var (
+	Cfg APIConfig
+	v   = viper.New()
+)
+
+func init() {
+	// Default config
+	ref := reflect.ValueOf(defaultConf)
+
+	// TODO: Figure out a better way to set default values - this will not work with nested keys
+	for i := 0; i < ref.NumField(); i++ {
+		v.SetDefault(ref.Type().Field(i).Tag.Get("mapstructure"), ref.Field(i).Interface())
+	}
+
+	// Flags
+	pflag.StringP("base_url", "b", "", "Address to which API will bind and start listening")
+	pflag.StringP("bind_address", "l", "", "Base URL (useful if being proxied through something like nginx). Value needs to be full URL up to the application (e.g. https://braize.pajlada.com/chatterino)")
+	pflag.Uint64("max_content_length", 5*1024*1024, "Max content size (in bytes) - requests with body bigger than this value will be skipped")
+	pflag.Bool("enable_lilliput", true, "When enabled, will attempt to use lilliput library for building animated thumbnails. Could increase memory usage by a lot.")
+	pflag.String("discord_token", "", "Discord token")
+	pflag.String("twitch_client_id", "", "Twitch client ID")
+	pflag.String("twitch_client_secret", "", "Twitch client secret")
+	pflag.String("youtube_api_key", "", "YouTube API key")
+	pflag.String("twitter_bearer_token", "", "Twitter bearer token")
+	pflag.String("imgur_client_id", "", "Imgur client ID")
+	pflag.String("oembed_facebook_app_id", "", "oEmbed Facebook app ID")
+	pflag.String("oembed_facebook_app_secret", "", "oEmbed Facebook app secret")
+	pflag.String("oembed_providers_path", "./providers.json", "Path to a json file containing supported oEmbed resolvers")
+	pflag.Parse()
+	v.BindPFlags(pflag.CommandLine)
+
+	// figure out XDG_DATA_CONFIG to be compliant with the standard
+	xdgConfigHome, exists := os.LookupEnv("XDG_CONFIG_HOME")
+	if !exists || xdgConfigHome == "" {
+		xdgConfigHome = fmt.Sprintf("$HOME/.config/%s/", appName)
+	}
+
+	// File
+	v.SetConfigName(appName)
+	v.SetConfigType("yaml")
+	v.AddConfigPath("/etc/")
+	v.AddConfigPath(xdgConfigHome)
+	v.AddConfigPath(".")
```

Comment:
- Linux: With new additions from 83d0083 and b77710f everything works perfectly (also tested custom `$XDG_CONFIG_HOME`)
- Windows: Made it work in 08c8ad4, furthermore I've made it use `%APPDATA%` as default config directory since that is the closest equivalent. I hope that using `runtime` package for that is fine. Or should we experiment with something like in `thumbnail` package?
- MacOS: Friend of mine tested everything on mac on 08c8ad4 and it seems to work :+1:
<details>
<summary>Testing on MacOS</summary>
Input:

```bash
mkdir $HOME/.config/chatterino-api
sudo mkdir /etc/chatterino-api

timeout 2 ./api 2> /dev/null

echo 'base-url: "my-custom-base-url"\nbind-address: ":5000"' | sudo tee /etc/chatterino-api/config.yaml
./api 2> /dev/null

echo 'bind-address: ":6000"' > ./config.yaml
./api 2> /dev/null

CHATTERINO_API_BIND_ADDRESS=:7000 ./api 2> /dev/null

CHATTERINO_API_BIND_ADDRESS=:7000 ./api --bind-address=:8000 2> /dev/null

echo 'base-url: "forsen-dot-tv"\nbind-address: ":2000"' > $HOME/.config/chatterino-api/config.yaml
./api 2> /dev/null
```

Output:

```
config.APIConfig{BaseURL:"", BindAddress:":1234", MaxContentLength: 0x500000, EnableLilliput:true, DiscordToken:"", TwitchClientID:"", TwitchClientSecret:"", YoutubeApiKey:"", TwitterBearerToken:"", ImgurClientID:"", OembedFacebookAppID:"", OembedFacebookAppSecret:"", OembedProvidersPath:"./providers.json"}
base-url: "my-custom-base-url"
bind-address: ":5000"
config.APIConfig{BaseURL:"my-custom-base-url", BindAddress:":5000", MaxContentLength: 0x500000, EnableLilliput:true, DiscordToken:"", TwitchClientID:"", TwitchClientSecret:"", YoutubeApiKey:"", TwitterBearerToken:"", ImgurClientID:"", OembedFacebookAppID:"", OembedFacebookAppSecret:"", OembedProvidersPath:"./providers.json"}
config.APIConfig{BaseURL:"my-custom-base-url", BindAddress:":6000", MaxContentLength: 0x500000, EnableLilliput:true, DiscordToken:"", TwitchClientID:"", TwitchClientSecret:"", YoutubeApiKey:"", TwitterBearerToken:"", ImgurClientID:"", OembedFacebookAppID:"", OembedFacebookAppSecret:"", OembedProvidersPath:"./providers.json"}
config.APIConfig{BaseURL:"my-custom-base-url", BindAddress:":7000", MaxContentLength: 0x500000, EnableLilliput:true, DiscordToken:"", TwitchClientID:"", TwitchClientSecret:"", YoutubeApiKey:"", TwitterBearerToken:"", ImgurClientID:"", OembedFacebookAppID:"", OembedFacebookAppSecret:"", OembedProvidersPath:"./providers.json"}
config.APIConfig{BaseURL:"my-custom-base-url", BindAddress:":8000", MaxContentLength: 0x500000, EnableLilliput:true, DiscordToken:"", TwitchClientID:"", TwitchClientSecret:"", YoutubeApiKey:"", TwitterBearerToken:"", ImgurClientID:"", OembedFacebookAppID:"", OembedFacebookAppSecret:"", OembedProvidersPath:"./providers.json"}
config.APIConfig{BaseURL:"forsen-dot-tv", BindAddress:":6000", MaxContentLength: 0x500000, EnableLilliput:true, DiscordToken:"", TwitchClientID:"", TwitchClientSecret:"", YoutubeApiKey:"", TwitterBearerToken:"", ImgurClientID:"", OembedFacebookAppID:"", OembedFacebookAppSecret:"", OembedProvidersPath:"./providers.json"}
```

</details>